mesonbuild/compilers/d.py

```diff
@@ -691,10 +711,11 @@ def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoic
                  exe_wrapper: T.Optional['ExternalProgram'] = None,
                  linker: T.Optional['DynamicLinker'] = None,
                  full_version: T.Optional[str] = None,
-                 is_cross: bool = False):
+                 is_cross: bool = False, version_output: str):
```

Comment:
You can't have arguments without default values after those with default values, so either this needs to have a default value, or it needs to be moved before the ones with default value.

I think I'd be tempted to make `find_ldc_dmd_frontend_version` return an `Optional[str]`, and allow this to be `Optional[str] = None`, then for old versions you can just check if the value is None.

Reply:
Actually, the more I think about it, the more I think it would be better to just return a bool and store that. Then you can just do:
```meson
if self._dmd_has_depfile:
   return ['<depfile args>']
return []
```

Reply:
In the Qt module, where we also need to determine whether depfiles are supported I set a class attribute rcc_supports_depfiles, and defined it to True in `__init__` if the version was new enough (otherwise logged a warning that dependencies would be unreliable).

Later in the function that handles rcc, I checked `self.rcc_supports_depfiles`.

So yes, I too would go this way.

Reply:
I think the GDC code does this for several features as well.